### PR TITLE
Np 45321 refactor fetch nvi candidate handler

### PR DESCRIPTION
**Pull request by monaullah**
- Use CandidateBO to fetch applicable candidate in stead of NviService
- Don't allow updating approval status or assignee on non applicable candidate

### CODE REVIEW

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/CandidateBO.java [hunk 1]
```diff
@@ -348,6 +343,34 @@ private static String mapToUsernameString(Username assignee) {
...
+    }
+
+    private static boolean isApplicable(CandidateDao candidateDao) {
+        return candidateDao.candidate().applicable();
+    }
+
+    private boolean isApplicable() {
+        return original.candidate().applicable();
+    }
+
```
Comment (truhacevkir) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/CandidateBO.java:373]: kanskje til en isNotApplicable metode? Synes fire metoder er litt overkill :)

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/CandidateBO.java [hunk 2]
```diff
@@ -348,6 +343,34 @@ private static String mapToUsernameString(Username assignee) {
         return assignee != null ? assignee.value() : null;
     }
 
+    private void validateCandidateState() {
+        if (periodStatus.status().equals(Status.CLOSED_PERIOD)) {
+            throw new IllegalStateException(PERIOD_CLOSED_MESSAGE);
+        }
```
Comment (truhacevkir) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/CandidateBO.java:349]: I know that i added this if statement, but it should be Statsus.CLOSED_PERIOD.equals(...

**Review by truhacevkir — APPROVED** [review]

File: nvi-test/src/main/java/no/sikt/nva/nvi/test/TestUtils.java [hunk 3]
```diff
@@ -55,6 +55,8 @@ public final class TestUtils {
     private static final String PUBLICATION_API_PATH = "publication";
     private static final String API_HOST = "example.com";
 
+    public static final int CURRENT_YEAR = LocalDate.now().getYear();
```
Comment (hackon) [inline nvi-test/src/main/java/no/sikt/nva/nvi/test/TestUtils.java:57]: Year.now()

**Review by hackon — APPROVED** [review]

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/CandidateBO.java [hunk 4]
```diff
@@ -49,15 +50,15 @@
 
 public final class CandidateBO {
 
-    public static final String PERIOD_CLOSED_MESSAGE = "Period is closed, perform actions on candidate is forbidden!";
-    public static final String PERIOD_NOT_OPENED_MESSAGE = "Period is not opened yet, perform actions on candidate is"
-                                                           + " forbidden!";
+    public static final String DELETE_MESSAGE_ERROR = "Can not delete message you does not own!";
```
Comment (hackon) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/CandidateBO.java:53]: Private?

**Review by hackon — APPROVED** [review]

**Pull request merged**